api/v1beta1/openstackdataplanenodeset_webhook.go

```diff
@@ -87,8 +94,54 @@ func (r *OpenStackDataPlaneNodeSet) ValidateCreate() error {
 // ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
 func (r *OpenStackDataPlaneNodeSet) ValidateUpdate(old runtime.Object) error {
 	openstackdataplanenodesetlog.Info("validate update", "name", r.Name)
+	oldNodeSet, ok := old.(*OpenStackDataPlaneNodeSet)
+	if !ok {
+		return apierrors.NewInternalError(
+			fmt.Errorf("Expected a OpenStackDataPlaneNodeSet object, but got %T", oldNodeSet))
+	}
+
+	var errors field.ErrorList
+	// Some changes to the baremetalSetTemplate after the initial deployment would necessitate
+	// a redeploy of the node. Thus we should block these changes and require the user to
+	// delete and redeploy should they wish to make such changes after the initial deploy.
+	// If the BaremetalSetTemplate is changed, we will offload the parsing of these details
+	// to the openstack-baremetal-operator webhook to avoid duplicating logic.
+	if !reflect.DeepEqual(r.Spec.BaremetalSetTemplate, oldNodeSet.Spec.BaremetalSetTemplate) {
+		// Initialize OpenStackBaremetalSet with old spec details
+		oldBaremetalSetObject := &baremetalv1.OpenStackBaremetalSet{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      r.Name,
+				Namespace: r.Namespace,
+			},
+		}
+		oldNodeSet.Spec.BaremetalSetTemplate.DeepCopyInto(&oldBaremetalSetObject.Spec)
+
+		// Initialize OpenStackBaremetalSet with new spec details
+		baremetalSetObject := &baremetalv1.OpenStackBaremetalSet{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      r.Name,
+				Namespace: r.Namespace,
+			},
+		}
+		r.Spec.BaremetalSetTemplate.DeepCopyInto(&baremetalSetObject.Spec)
+
+		// Call openstack-baremetal-operator ValidateUpdate() webhook to parse changes
+		err := baremetalSetObject.ValidateUpdate(oldBaremetalSetObject)
```

Comment:
Does this work? We're creating a new OpenStackBaremetalSet object and calling ValidateUpdate() on it. So I think the ownerUIDLabelSelector won't match when selecting existing BMHs[1].

[1] https://github.com/openstack-k8s-operators/openstack-baremetal-operator/blob/main/api/v1beta1/openstackbaremetalset_webhook.go#L136-L141